### DISCUSSION TITLE
Outputting streams to a text file in chunks

### DISCUSSION THREAD

**Discussion by mickvikt** [paillave/Etl.Net#475] [Q&A]
I have a question about chunking a long stream before outputting it to a text file.
`Chunk` operator returns `IStream<Enumerable<TIn>>`, but extension method `ToTextFileValue` accepts ` IStream<TIn>`.

How do I transform  `IStream<Enumerable<TIn>>` to  ` IStream<TIn>` to be able to output to file?

Thank you.

**Comment by paillave** [comment]
Let me get it straight: from one stream of something you want to issue several files of X lines?

Anyway, this [page about file creation](https://paillave.github.io/Etl.Net/docs/recipes/createFiles) might help you. Let me know otherwise.

**Reply by mickvikt** [reply]
Not exactly, I have a long data stream from the database query and I want to output it to one text file.
The problem is that I get Stream too long or Out of memory errors.
I thought that chunking the stream could mitigate this problem.

**Comment by paillave** [comment]
Ah ok, I'll come back to with an example. It's an interesting use case 🙂

________________________________
